sdk/javascript/src/utils/errors.js

```
@@ -0,0 +1,132 @@
+export function createErrorFromResponse(response) {
+  const { statusCode, message, error, details } = response;
+
+  switch (statusCode) {
+    case 401:
+      return new AuthenticationError(message || 'Authentication failed', details);
+    case 403:
+      return new AuthorizationError(message || 'Access denied', details);
+    case 400:
+      return new ValidationError(message || 'Validation failed', details);
+    case 429:
+      const retryAfter = parseInt(response.headers?.['retry-after'] || '60');
```

Biome (2.1.2)

[error] Other switch clauses can erroneously access this declaration.
Wrap the declaration in a block to restrict its access to the switch clause.

The declaration is defined in this switch clause:

Safe fix: Wrap the declaration in a block.


(lint/correctness/noSwitchDeclarations)